### PR TITLE
Run integration tests only in ccov job

### LINKED ISSUE #397: We are running integration tests twice
https://github.com/fedimint/fedimint/pull/383#issuecomment-1218138538

### PR DESCRIPTION
**Pull request by dpc**
Also run them on one thread only.

Fix #397

### CODE REVIEW

**Comment by justinmoon** [comment]
It seems to me that this no longer runs `rust-tests.sh`. One important thing that `run-tests.sh` does is run `bitcoind` and `lightningd` fixtures. Tests still pass, but they just mocking out `bitcoind` and `lightningd`. See `MINIMINT_TEST_REAL` environment variable.

**Comment by dpc** [comment]
@justinmoon The "Test workspace" step is `cargo test --profile release` on the whole workspace, which AFAIU executes integration tests as well. The reason there is no output from this test is because it passed before and Nix doesn't bother running it (cachix).

**Comment by dpc** [comment]
> I think it is still being run:
> - [here](https://github.com/fedimint/fedimint/pull/401/files#diff-fb85e3427e423c8f80d79fb275a6683ccff2fe4340150528e54e389ea01445afL73)  it refers `flakes .#cli-test.rust-tests`
> - which is in flakes [here](https://github.com/fedimint/fedimint/blob/master/flake.nix#L355-L360) and will run [this](https://github.com/fedimint/fedimint/blob/master/flake.nix#L230)
> 
> @justinmoon @dpc 

It was disabled atm of @justinmoon comment. I've fixed it (hopefully) since then.

File: .github/workflows/ci-nix.yml [hunk 1]
```diff
@@ -68,7 +69,7 @@ jobs:
         # run: nix-shell --command ./scripts/clientd-tests.sh
         run: nix build -L --extra-experimental-features nix-command --extra-experimental-features flakes .#cli-test.clientd
 
-      - name: Integration Tests
+      - name: Integration Tests (no fixtures)
```
Comment (justinmoon) [inline .github/workflows/ci-nix.yml:72]: I'm confused by this "fixtures" terminology as mentioned in https://github.com/fedimint/fedimint/pull/408. I presume this means we're using bitcoind and lightningd but I had to think about it ....

**Comment by justinmoon** [comment]
> @justinmoon The "Test workspace" step is `cargo test --profile release` on the whole workspace, which AFAIU executes integration tests as well. The reason there is no output from this test is because it passed before and Nix doesn't bother running it (cachix).

"Test workspace" runs `.#workspaceTest`, which is this:

```
        workspaceTest = craneLib.cargoBuild (commonArgs // {
          cargoArtifacts = workspaceBuild;
          doCheck = true;
        });
```

I don't see `cargo test --profile release` being run anywhere in our project ...

**Comment by dpc** [comment]
`doCheck = true;` means that `cargo test --profile release` will be run on the workspace. https://github.com/ipetkov/crane/blob/master/docs/API.md#optional-attributes , see `cargoCheckCommand` and `doCheck`.

You can verify locally with `nix build -L .#workspaceTest` (in case it does nothing, make some changes in the test code to invalidate cache).

Now I have disabled it because:

`env RUST_TEST_THREADS=1 cargo llvm-cov --workspace --lcov --output-path $out/lcov.info"`

is running the same tests (unit + integration). Per https://github.com/taiki-e/cargo-llvm-cov#basic-usage:

> By default, run tests (via cargo test), and print the coverage summary to stdout. 

To verify: `nix build -L .#workspaceCov`. (waiting for it to finish myself to double check).

**Review by justinmoon — APPROVED** [review]

**Pull request merged**